assets/js/search.js

```diff
@@ -247,14 +268,103 @@
+
+            const link = node.querySelector('a');
+            if(link){
+                logUbiEvent('item_click', link);
+            }
+
+            return true;
+        };
+
         const navToHighlightedResult = () => {
             const searchResultClassName = 'top-banner-search--field-with-results--field--wrapper--search-component--search-results--result';
             elResults.querySelector(`.${searchResultClassName}.highlighted a[href]`)?.click?.();
         };
 
+        /**
+         * Find item and position clicked
+         * Modifies the ubi event data if the item is found
+         * @param {*} ubiEvent
+         * @param {*} link 
+         * @returns 
+         */
+        const setUbiClickData = (ubiEvent, link) => {
+            ubiEvent.event_attributes.position = new UBI.UbiPosition({x:link.offsetLeft, y:link.offsetTop});
+
+            if(link.hasAttribute('id')){
+                let id = link.id;
+                //try to find the item ordinal within the result list
+                let resultIds = sessionStorage.getItem('result_ids');
+                if(resultIds != null && resultIds.length > 0){
+                    resultIds = resultIds.split(',');
+                    let ordinal = resultIds.findIndex( i => i===id );
+                    //if found, ordinal starts at 1
+                    if(ordinal != -1){
+                        ubiEvent.event_attributes.position.ordinal = ordinal + 1;
+                        if(ubiEvent.message == undefined || ubi_event.message == null){
+                            ubiEvent.message = `Clicked item ${ordinal+1} out of ${resultIds.length}`
+                        }
+                        
+                        try{
+                            let searchResults = JSON.parse(sessionStorage.getItem('search_results'));
+                            let obj = searchResults[id];
+                            if(obj != null){
+                                ubiEvent.event_attributes.object = obj;
+                                ubiEvent.event_attributes.position.trail = getBreadcrumbs(obj);
+                            }
+                        }catch(e){
+                            console.warn(e);
+                        }
+                    }
+                }
+            }
+            return ubiEvent;
+        };
+
+
+        const makeUbiEvent = (name, event_type, data) => {
+            let e = new UBI.UbiEvent(name);
+            let t = TimeMe.getTimeOnPageInSeconds(window.location.pathname);
+            if(t != null){
+                e.event_attributes['dwell_seconds'] = t;
+            }
+
+            if(name == 'search'){
+                e.message_type = 'QUERY';
+                e.message = data.search_term;
+                e.event_attributes.object = data;
+            } else if(name == 'item_click') {
+                e = setUbiClickData(e, data);
+            } else {
+                switch(event_type) {
```

Comment:
Likewise I'd like to see this switch statement filled in or removed....   To get this to something that is committable.